features/ui/button/button.tsx

```diff
@@ -18,3 +18,260 @@ export const Button = styled.button`
     padding: 0;
   }
 `;
+export const ButtonStyle = styled.button<{
+  size: ButtonSize;
+  color: ButtonColor;
+  state: ButtonState;
+  icon: ButtonIcon;
+}>`
+  cursor: pointer;
+  border-radius: 8px;
+  color: #fff;
+  display: flex;
+  align-items: center;
+  box-sizing: border-box;
+
+  // remove default button styles
+  border: none;
+  margin: 0;
+  padding: 0;
+  background: transparent;
+  line-height: normal;
+  -webkit-font-smoothing: inherit;
+  -moz-osx-font-smoothing: inherit;
+  -webkit-appearance: none;
+
+  &::-moz-focus-inner {
+    border: 0;
+    padding: 0;
+  }
+
+  ${(props) => {
+    switch (props.size) {
+      case ButtonSize.sm:
+        return css`
+          padding: 8px 14px;
+          ${textFont("sm", "medium")}
+        `;
+      case ButtonSize.md:
+        return css`
+          padding: 10px 16px;
+          ${textFont("sm", "medium")}
+        `;
+      case ButtonSize.lg:
+        return css`
+          padding: 10px 18px;
+          ${textFont("md", "medium")}
+        `;
+      case ButtonSize.xl:
+        return css`
+          padding: 12px 20px;
+          ${textFont("md", "medium")}
+        `;
+    }
+  }}
+
+  ${(props) =>
+    props.state == "default"
+      ? props.color == "secondary"
```

Comment:
A switch statement might be more readable. Alternatively I'd prefer `if` statements with `returns` over these nested ternaries